Monal/Classes/MLIQProcessor.m

```diff
@@ -784,6 +785,20 @@ +(BOOL) processRosterWithAccount:(xmpp*) account andIqNode:(XMPPIQ*) iqNode
     }];
 $$
 
+$$class_handler(handleChangePassword, $$ID(xmpp*, account), $$ID(NSString*, newPass), $$ID(NSUUID*, uuid))
+    DDLogDebug(@"Handling password change");
+    [[MLXMPPManager sharedInstance] updatePassword:newPass forAccount:account.accountID];
+    MLPromise* promise = [account.promiseRegistry getPromise:uuid];
+    [promise resolve:nil];
+$$
+
+$$class_handler(handleChangePasswordInvalidation, $$ID(xmpp*, account), $$ID(NSUUID*, uuid))
+    DDLogDebug(@"Invalidating password change");
+    NSError* error = [NSError errorWithDomain:@"Monal" code:0 userInfo:@{NSLocalizedDescriptionKey: @"Could not change password"}];
+    MLPromise* promise = [account.promiseRegistry getPromise:uuid];
+    [promise resolve:error];
```

Comment:
I tried splitting `resolve` into `fulfill` and `reject`, but I ended up with two functions and variables in `MLPromise` that seemed to do mostly the same thing. Do you think that just having `resolve` and passing the error is explicit enough? Happy to do it either way, but this way seemed more natural to me.

Reply:
Well, that's because the AnyPromise API is overloaded to accept any object and automatically transitions the state to "rejected", if its an instance of NSError (and "fulfilled" if its something different).

Honestly I don't like that design and the Swift part of PromiseKit (the promise lib we are using) has dedicated methods for rejecting and fulfilling.

If, in the long term, the majority of promises used in the objc codebase will be MLPromise objects, I'd like to provide dedicated methods for fulfill and reject (possibly adding an assertion to check if the data provided to reject is an instance of NSError). You are right, internally we could still use only one ivar to hold this data regardless if it was provided calling the fulfill or reject methods.

Reply:
Hmmm, I think we could improve that API further.
What about abstracting the uuid away and somehow directly provide the MLHandler as argument to the handler? We should discuss this in chat.